### ISSUE TITLE
Hyper-V enabled on Windows 10 Home, verified.

### ISSUE THREAD

**Issue by Robert-Becker** [MicrosoftDocs/Virtualization-Documentation#915]
Please update your thread. 
Tested and verified on 11/21/2018. 
You can run Hyper-V on both Windows 10 HOME and PRO additions. 

Step to enable this option can be found at the following link. 
https://blogs.technet.microsoft.com/canitpro/2015/09/08/step-by-step-enabling-hyper-v-for-use-on-windows-10/

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1df65b2d-24ba-71b0-0648-14c21f5b1b26
* Version Independent ID: a89c84df-6ef7-66a9-eec1-35ba94afdb1b
* Content: [Enable Hyper-V on Windows 10](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v)
* Content Source: [virtualization/hyper-v-on-windows/quick-start/enable-hyper-v.md](https://github.com/Microsoft/Virtualization-Documentation/blob/live/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v.md)
* Product: **windows-10-hyperv**
* GitHub Login: @scooley

**Comment by jackpoz** [comment]
@Robert-Becker how did you install Hyper-V on Windows 10 Home ?

**Comment by w-biggs** [comment]
How does that link indicate in any way that Hyper-V works on Win10 Home?

**Comment by Robert-Becker** [comment]
Hyper-V is installed natively on Windows 10 home all you have to do is activate it.(Default is turned off) I am coming from a win7 with the free upgrade to win10home. I am not sure if that makes a difference. I don't have my screen shots on this computer anymore but I used that link, had 2 different VM instances running at the same time and I'm using win10home. If you enter the command "systeminfo" in your command prompt at the bottom you'll see a section called "Hyper-V Requirements" I had a few turned off by default. Turned them on using that link and now life is good. Just do that and you'll be fine. I'm using oracle box for my VMs but you can use whatever makes you comfortable I just chose it because I hadn't used it before. With all that being said.... hardware is hardware if you don't have good enough hardware you can't use Hyper-V or VMs so hopefully you're not trying to party like its 1999. Best of Luck!

**Comment by jamiesondunne** [comment]
The link literally says repeatedly that hyper-V is only available on the "pro , and "enterprise" editions

**Comment by jamiesondunne** [comment]
![image](https://user-images.githubusercontent.com/11201336/56003905-05e19e00-5c97-11e9-81e3-37c4572ab137.png)

**Comment by w-biggs** [comment]
I incredibly highly doubt you actually got it working on Home. Can you paste your results when running `systeminfo`?

**Comment by jackpoz** [comment]
@Robert-Becker could you write here the steps to enable it on Home, even better with screenshots ?

**Comment by Robert-Becker** [comment]
Guys... seriously.... did you run the command in command prompt? Did you see "hyper-v requirements"? If you did... did you follow the steps in that link I posted? If you did then we wouldn't be having this conversation right now. 

Jaime, bravo on reading 2 sentences and giving up. You'll be a tech master in no time with that attitude! 
Also once you follow the steps and realize what you just posted is wrong, kindly remove it so others don't think read it and believe it like you did. 

For the rest of you, Here is a screenshot of my OS showing win10home, showing hyper-v requirements section, and 2 different oracle vms running in the background on is win2008svr the other is win2008svrcore only.... 

One last time.... if your hardware is garbage... no amount of firmware/software will ever make it work. 

Type 'Systeminfo' into your command prompt if you don't see hyper-v requirements then you don't get to have fun. If you do see the requirements section, follow the steps in the link and you'll be ok. 

I won't be posting to this thread anymore. You guys have all the information you need to make it work. If it doesn't work you didn't do something right or your hardware is garbage. Good luck!

![Hyperv screenshot](https://user-images.githubusercontent.com/45242351/56055696-260d6d80-5d28-11e9-95cb-e22a28c34cc7.PNG)

**Comment by jackpoz** [comment]
![image](https://user-images.githubusercontent.com/1153754/56058826-d0a28200-5d62-11e9-9f23-f4a96379dd65.png)

![image](https://user-images.githubusercontent.com/1153754/56058838-dac48080-5d62-11e9-998e-6bea0b190327.png)

![image](https://user-images.githubusercontent.com/1153754/56058867-f16ad780-5d62-11e9-8233-7695d65690c2.png)

I meet all the requirements but there's no Hyper-V available in "Turn Windows features on or off". The link you posted doesn't say anything about how to make that option appear, any idea ?

**Comment by GranoblasticMan** [comment]
It sounds like @Robert-Becker is confused. If Hyper-V is enabled, VirtualBox will NOT run. They're two separate hypervisor systems, and Hyper-V locks exclusivity to the virtualization features at boot-time.

**Comment by Robert-Becker** [comment]
I believe Grano is partially correct but we're mincing words here. I'm technically not using hyper-v manager to run my VMs. With that being said, my VM's didn't work until I enabled the hyper-v features shown in my picture. For those of us who are looking to play with VMs without upgrading this is still your path. 

Jackpoz if you followed all the steps the best you could, you can now use a VM program like virtual box or VMware if your hardware is good enough.

**Comment by w-biggs** [comment]
So you didn't have Hyper-V enabled. Hyper-V is necessary for many things, including running Docker, and *is* impossible on W10 Home. It's not mincing words, you're conflating two very different things.

Whoever's able to close this issue should do so.

**Comment by dxgldotorg** [comment]
Is this a case of Hyper-V's core being enabled to run certain Windows features like virtualization based security?

**Comment by waleedelsafty** [comment]
> I meet all the requirements but there's no Hyper-V available in "Turn Windows features on or off". The link you posted doesn't say anything about how to make that option appear, any idea ?
> 
> Even trying other ways explained at https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v doesn't work:
> ![image](https://user-images.githubusercontent.com/1153754/56059207-df3d6900-5d63-11e9-9ad0-d3bd5af2e3fe.png)
 it is true  that  windows hoe edition is not  equipped to have hyper v on it, and by default, it  doesn't have it installed  and its prerequisites are not enabled, simply because windows home is a very simple version of windows and it  doesn't assume  you will need to do much work on it  except  some schoolwork  entertainment  and basic ms office  stuff,   but  that doesn't mean it can not handle it if you turn the right keys on, and add  the missing packages  
it is  actually  very possible to run hyper v  on windows 10 home edition 

 the checks  you made  to ensure  you have all the requirements   for hyper v  is correct,
and you already have all the requirements  for it 

There are four basic requirements for Hyper-V to be installed on a Windows 10 computer:

    VM Monitor Mode Extensions
    Virtualization enabled in firmware
    Second level address translation
    Data execution prevention
and  you have  done that already
 
> ![image](https://user-images.githubusercontent.com/1153754/56058838-dac48080-5d62-11e9-998e-6bea0b190327.png)
> 

 but your problem is that  after  you  had all the prerequisites for hyper v to work < but you did not   add the packages  for hyper-v  so  you could not  see it in the "Turn Windows features on or off"   simply  because it still does not exist  on your machine 

you can get  all the packages   into your  windows system   by  running a simple batch file 
create a new batch file in notepad   with the following code 

`pushd "%~dp0"
dir /b %SystemRoot%\servicing\Packages\*Hyper-V*.mum >hyper-v.txt
for /f %%i in ('findstr /i . hyper-v.txt 2^>nul') do dism /online /norestart /add-package:"%SystemRoot%\servicing\Packages\%%i"
del hyper-v.txt
Dism /online /enable-feature /featurename:Microsoft-Hyper-V -All /LimitAccess /ALL
pause`

name it 
whatever you want "HYPER-V Package Installer.bat" for example
right-click on the file and select Run as Administrator. This will trigger the installation script. It may take some time to complete the installation. Please let it complete without interruption.
![Screenshot 2020-04-11 07 54 32](https://user-images.githubusercontent.com/51638781/79035754-230ebc80-7bca-11ea-8bf7-1556bd0ebb7e.png)

it will start downloading and installing all the required package for hyper-v to work on your machine, it will take from  3 to 15 minutes downloading and installing depending on your connection speed. 
![Running-enable-hyper-V-batch-installation-file](https://user-images.githubusercontent.com/51638781/79035824-acbe8a00-7bca-11ea-83a7-537db7ac1a9e.jpg)
after  it is finished,   restart  your computer  
 it will  do a very quick update      couple of minutes at most,  don't interrupt it 
![Working-on-Updates-installing-Hyper-V](https://user-images.githubusercontent.com/51638781/79035840-de375580-7bca-11ea-9600-26df554972ea.jpg)

and  after your  machine restarts   try again  the  "Turn Windows features on or off"  and  it will magically appear  on the list 

this is  the before :

![Screenshot 2020-04-11 07 19 41](https://user-images.githubusercontent.com/51638781/79035884-2bb3c280-7bcb-11ea-86a3-d8b57d70c687.png)



and this is the after : 
![Screenshot 2020-04-11 07 56 17](https://user-images.githubusercontent.com/51638781/79035887-3706ee00-7bcb-11ea-9baa-1bb386580eed.png)

**Comment by Uraharadono** [comment]
@waleedelsafty 

I have tried your solution to no avail. Then I searched a bit over internet and came across solution that is based on creating .bat as well, but for whatever reason it works. 
(Don't hate me, I am in rush to finish work related stuff, and could not be bothered to check why your explanation doesn't work)


Link to paste with explanations: https://pastebin.com/cX6nupy4

**Comment by ilikeballoons** [comment]
> some time to complete the installation. Please let it complete without interruption.
> ![Screenshot 2020-04-11 07 54 32](https://user-images.githubusercontent.com/51638781/79035754-230ebc80-7bca-11ea-8bf7-1556bd0ebb7e.png)

You should have probably reviewed one item in the context menu in this screenshot before you posted it :)

**Comment by skdursh** [comment]
> How does that link indicate in any way that Hyper-V works on Win10 Home?

If I remember correctly it used to be available on the Home edition when it was still in beta testing, but was made a "pro" feature in a relatively recent update. You can still enable it on Windows 10 Home, but it takes a couple extra "legally grey area" steps.

**Comment by dxgldotorg** [comment]
Looking at the unhacked OptionalFeatures dialog you can see that some features that use the Hyper-V platform at a low level like Virtual Machine Platform and Windows Hypervisor Platform are already available indicating the hypervisor is available in all SKUs of Windows 10.  It's just the Client Hyper-V functionality that allows creation and operation of virtual machines without using an external virtualization utility or WSL2 is the part exclusive to Professional and up.

**Comment by TyIsI** [comment]
It looks like that with the latest feature release of Win10 Home (2004/19041) Hyper-V components have been enabled.

This is great, because now WSL works on Windows Home. However, running WSL and VirtualBox has now become mutually exclusive. VirtualBox - and I assume similar virtualization technologies - does not work unless `hypervisorlaunchtype` has been set to off, which consequently gets in the way of using WSL.

Hyper-V is not available as a separate option in `Windows Features`, but I'm running with `Virtual Machine Platform`, `Windows Hypervisor Platform` and `Windows Subsystem for Linux` enabled.

If it helps, there are multiple events logs in Event Viewer for Hyper-V (Compute, Guest-Drivers, Hypervisor, StorageVSP, VID, VmSwitch and Worker).

**Comment by dxgldotorg** [comment]
@Tylsl does your Home edition have Windows Hypervisor Platform?  If so, have you tried enabling that?

**Comment by SayakMukhopadhyay** [comment]
I am running a similar setup as @TyIsI . I have `Virtual Machine Platform`, `Windows Hypervisor Platform` and `Windows Subsystem for Linux` enabled. Moreover, I have Docker for Windows enabled and running. I also have VirtualBox 6.1.12 installed and a couple of Ubuntu guests. They start off fine but I face checksum mismatch issues whenever I try to install anything via `apt` or `npm`.

**Comment by TyIsI** [comment]
@dxgldotorg I do have that enabled:

![image](https://user-images.githubusercontent.com/3118952/89702430-125f2080-d8f6-11ea-9389-eb12a99097c0.png)

I just tried to create a new virtual machine in Virtual Box (6.0.22 r137980 (Qt5.6.2)), but got the following error:

![image](https://user-images.githubusercontent.com/3118952/89702480-86012d80-d8f6-11ea-91d2-832d0df8bc67.png)

```
Result Code: | E_FAIL (0x80004005)
-- | --
Component: | ConsoleWrap
Interface: | IConsole {872da645-4a9b-1727-bee2-5585105b9eed}
```

**Comment by windows10win7posreadywoeskio** [comment]
> @dxgldotorg I do have that enabled:
> 
> ![image](https://user-images.githubusercontent.com/3118952/89702430-125f2080-d8f6-11ea-9389-eb12a99097c0.png)
> 
> I just tried to create a new virtual machine in Virtual Box (6.0.22 r137980 (Qt5.6.2)), but got the following error:
> 
> ![image](https://user-images.githubusercontent.com/3118952/89702480-86012d80-d8f6-11ea-91d2-832d0df8bc67.png)
> 
> ```
> Result Code: | E_FAIL (0x80004005)
> -- | --
> Component: | ConsoleWrap
> Interface: | IConsole {872da645-4a9b-1727-bee2-5585105b9eed}
> ```



Don't try to get HyperV on if you have VirtualBox.

**Comment by windows10win7posreadywoeskio** [comment]
> @dxgldotorg I do have that enabled:
> 
> ![image](https://user-images.githubusercontent.com/3118952/89702430-125f2080-d8f6-11ea-9389-eb12a99097c0.png)
> 
> I just tried to create a new virtual machine in Virtual Box (6.0.22 r137980 (Qt5.6.2)), but got the following error:
> 
> ![image](https://user-images.githubusercontent.com/3118952/89702480-86012d80-d8f6-11ea-91d2-832d0df8bc67.png)
> 
> ```
> Result Code: | E_FAIL (0x80004005)
> -- | --
> Component: | ConsoleWrap
> Interface: | IConsole {872da645-4a9b-1727-bee2-5585105b9eed}
> ```



Well, i think you should try enabling Hypervisor Off.

**Comment by TyIsI** [comment]
> Well, i think you should try enabling Hypervisor Off.

Disabled Windows Hypervisor Platform. WSL works, but VirtualBox did not.

Then upgraded VirtualBox to 6.1.12 r139181 (Qt5.6.2).
This fixed it.

I should have upgraded to the last version first, because after checking the changelog for VirtualBox, it did mention that for VirtualBox 6.1.4 (released February 19 2020): "Windows host: Restore the ability to run VMs through Hyper-V, at the expense of performance"
(I'm suspicious of another software tool installing an outdated version of VirtualBox during its installation and assumed - yeah, I know - that I had been on the most recent version.)

In short, if anyone is having issues with VirtualBox, make sure to upgrade to the most recent version.

**Comment by SayakMukhopadhyay** [comment]
> Then upgraded VirtualBox to 6.1.12 r139181 (Qt5.6.2).
> This fixed it.

Yeah, I am running the same version of VirtualBox and my OS boots up but I get checksum errors whenever I try to use apt-get or npm.

**Comment by TyIsI** [comment]
> Yeah, I am running the same version of VirtualBox and my OS boots up but I get checksum errors whenever I try to use apt-get or npm.

Let me see if this also occurs for me.

**Comment by geckoflume** [comment]
@TyIsI @SayakMukhopadhyay I am also running Windows 10 Home 2004, and I can't get WSL2 running next to Virtualbox (or VMware Workstation)... Did you find a workaround?

**Comment by dxgldotorg** [comment]
For me, VirtualBox only runs in "turtle" mode.

For VMWare, make sure you have at least version 16 which has Windows Hypervisor Platform support.  15.5 had an update to add it, but its implementation is incomplete.

**Comment by TyIsI** [comment]
I can confirm that this worked for me and that I've also been able to run `apt upgrade` without issue.

**Comment by Timelessprod** [comment]
> > I meet all the requirements but there's no Hyper-V available in "Turn Windows features on or off". The link you posted doesn't say anything about how to make that option appear, any idea ?
> > Even trying other ways explained at https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v doesn't work:
> > ![image](https://user-images.githubusercontent.com/1153754/56059207-df3d6900-5d63-11e9-9ad0-d3bd5af2e3fe.png)
> > it is true  that  windows hoe edition is not  equipped to have hyper v on it, and by default, it  doesn't have it installed  and its prerequisites are not enabled, simply because windows home is a very simple version of windows and it  doesn't assume  you will need to do much work on it  except  some schoolwork  entertainment  and basic ms office  stuff,   but  that doesn't mean it can not handle it if you turn the right keys on, and add  the missing packages
> > it is  actually  very possible to run hyper v  on windows 10 home edition
> 
> the checks you made to ensure you have all the requirements for hyper v is correct, and you already have all the requirements for it
> 
> There are four basic requirements for Hyper-V to be installed on a Windows 10 computer:
> 
> ```
> VM Monitor Mode Extensions
> Virtualization enabled in firmware
> Second level address translation
> Data execution prevention
> ```
> 
> and you have done that already
> 
> > ![image](https://user-images.githubusercontent.com/1153754/56058838-dac48080-5d62-11e9-998e-6bea0b190327.png)
> 
> but your problem is that after you had all the prerequisites for hyper v to work < but you did not add the packages for hyper-v so you could not see it in the "Turn Windows features on or off" simply because it still does not exist on your machine
> 
> you can get all the packages into your windows system by running a simple batch file create a new batch file in notepad with the following code
> 
> `pushd "%~dp0" dir /b %SystemRoot%\servicing\Packages\*Hyper-V*.mum >hyper-v.txt for /f %%i in ('findstr /i . hyper-v.txt 2^>nul') do dism /online /norestart /add-package:"%SystemRoot%\servicing\Packages\%%i" del hyper-v.txt Dism /online /enable-feature /featurename:Microsoft-Hyper-V -All /LimitAccess /ALL pause`
> 
> name it whatever you want "HYPER-V Package Installer.bat" for example right-click on the file and select Run as Administrator. This will trigger the installation script. It may take some time to complete the installation. Please let it complete without interruption. ![Screenshot 2020-04-11 07 54 32](https://user-images.githubusercontent.com/51638781/79035754-230ebc80-7bca-11ea-8bf7-1556bd0ebb7e.png)
> 
> it will start downloading and installing all the required package for hyper-v to work on your machine, it will take from 3 to 15 minutes downloading and installing depending on your connection speed. ![Running-enable-hyper-V-batch-installation-file](https://user-images.githubusercontent.com/51638781/79035824-acbe8a00-7bca-11ea-83a7-537db7ac1a9e.jpg) after it is finished, restart your computer it will do a very quick update couple of minutes at most, don't interrupt it ![Working-on-Updates-installing-Hyper-V](https://user-images.githubusercontent.com/51638781/79035840-de375580-7bca-11ea-9600-26df554972ea.jpg)
> 
> and after your machine restarts try again the "Turn Windows features on or off" and it will magically appear on the list
> 
> this is the before :
> 
> ![Screenshot 2020-04-11 07 19 41](https://user-images.githubusercontent.com/51638781/79035884-2bb3c280-7bcb-11ea-86a3-d8b57d70c687.png)
> 
> and this is the after : ![Screenshot 2020-04-11 07 56 17](https://user-images.githubusercontent.com/51638781/79035887-3706ee00-7bcb-11ea-9baa-1bb386580eed.png)

This solution didn't worked for me. The bat script does not work and return an "invalid filename or directory name syntax" on both CMD and PowerShell, both in admin mode. I sure have the 4 "yes" on `systeminfo`. The script only generated an empty text file on my desktop.

Any idea ? Thanks

**Comment by cyrusfirheir** [comment]
@Timelessprod (and anyone else trying this), the script just needs some cleaning up ;p

```bat
dir /b "%SystemRoot%\servicing\Packages\*Hyper-V*.mum" > hyper-v.txt
for /f %%i in ('findstr /i . hyper-v.txt 2^>nul') do dism /online /norestart /add-package:"%SystemRoot%\servicing\Packages\%%i"
del hyper-v.txt
Dism /online /enable-feature /featurename:Microsoft-Hyper-V -All /LimitAccess /ALL
pause
```

Tested. Works.
